operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -46,25 +47,51 @@ const (
 // This type exists so there's no chance of pushing random strings to our progress status
 type ProgressState string
 
+const (
+	cassandra_3_11_6 = "datastax/cassandra-mgmtapi-3_11_6:v0.1.2"
+	cassandra_4_0_0  = "datastax/cassandra-mgmtapi-4_0_0:v0.1.2"
+	dse_6_8_0        = "datastax/dse-server:6.8.0"
+	envBaseImageOs   = "BASE_IMAGE_OS"
+)
+
 // getImageForServerVersion tries to look up a known image for a server type and version number.
 // In the event that no image is found, an error is returned
 func getImageForServerVersion(server, version string) (string, error) {
-	const (
-		cassandra_3_11_6 = "datastax/cassandra-mgmtapi-3_11_6:v0.1.2"
-		cassandra_4_0_0  = "datastax/cassandra-mgmtapi-4_0_0:v0.1.2"
-		dse_6_8_0        = "datastax/dse-server:6.8.0"
-	)
-	sv := server + "-" + version
+	baseImageOs := os.Getenv(envBaseImageOs)
+
+	var imageCalc func(string) (string, bool)
+	var img string
+	var success bool
+
+	if baseImageOs == "" {
```

Comment:
but even alpine wouldn't be an empty string?

Reply:
so we re introduce the arg in the last stage of the dockerfile with a default of `""` so alpine is the initial default, then the `from` will use it in stage 2, then it will go back to `""` if its not manually set.

This way our default is always `""` whether its alpine or anything else.

Reply:
this is pretty clever! 👍